src/store/singing.ts

```diff
@@ -597,91 +599,82 @@ export const singingStore = createPartialStore<SingingStoreTypes>({
     },
   },
 
-  SET_SELECTED_NOTES: {
-    mutation(state, { noteIndices }: { noteIndices: Set<number> }) {
+  ADD_NOTE: {
+    mutation(state, { note }: { note: Note }) {
       if (state.score) {
-        state.selectedNotes = noteIndices;
+        state.selectedNoteIds.push(note.id);
+        state.score.notes = [...state.score.notes, note].sort(
+          (a, b) => a.position - b.position
+        );
       }
     },
-    async action(
-      { state, commit },
-      { noteIndices }: { noteIndices: Set<number> }
-    ) {
+    // ノートを追加する
+    // NOTE: 重複削除など別途追加
+    async action({ state, commit, dispatch }, { note }) {
       if (state.score === undefined) {
         throw new Error("Score is not initialized.");
       }
-      commit("SET_SELECTED_NOTES", { noteIndices });
-    },
-  },
-
-  CLEAR_SELECTED_NOTES: {
-    mutation(state) {
-      if (state.score) {
-        state.selectedNotes.clear();
+      if (!isValidNote(note)) {
+        throw new Error("The note is invalid.");
       }
-    },
-    async action({ commit }) {
-      commit("CLEAR_SELECTED_NOTES");
+      commit("ADD_NOTE", { note });
+      dispatch("RENDER");
     },
   },
 
-  ADD_NOTE: {
-    mutation(state, { note }: { note: Note }) {
+  UPDATE_NOTE: {
+    mutation(state, { note, index }: { note: Note; index: number }) {
       if (state.score) {
-        const notes = state.score.notes.concat(note).sort((a, b) => {
-          return a.position - b.position;
-        });
-        const addIndex = notes.indexOf(note);
-        state.selectedNotes.add(addIndex);
-        state.score.notes = notes;
+        state.score.notes.splice(index, 0, note);
```

Comment:
ここは動作を考えると以前のコードで問題ないと思います！
（何か意図があれば教えていただけると…！）
```suggestion
        const notes = [...state.score.notes];
        notes.splice(index, 1, note);
        state.score.notes = notes;
```